lib/funky/page.rb

```diff
@@ -0,0 +1,53 @@
+module Funky
+  class Page
+    extend DataParser
+
+    attr_reader :data
+
+    def initialize(data)
+      @data = data
+    end
+
+    # @return [String] the video ID.
+    def id
+      data[:id]
+    end
```

Comment:
@bruce-fs It's good that you extracted the common part of `Video` and `Page` into an external file.

But as you can see, there are even more repetitions, like `def initialize` and `attr_reader :data` and `def id`.

To avoid all repetitions, you are better off having `Funky::Page` and `Funky::Video` both *inherit* from a base abstrat class, rather than extending them with a module.